src/schema/rules/directories.yaml

```diff
@@ -0,0 +1,96 @@
+---
+# This file defines layouts of directories.
+# A directory may have a name, entity or value.
+#   - name: The directory name has exactly this value.
+#   - entity: The directory name takes the form of {key}-{value} for the specified entity
+#   - value: The directory name takes the form of {value} for the specified term (e.g., datatype)
+#
+# If opaque is true, then the contents of this directory are unspecified.
+# Level indicates whether a directory matching the naming scheme is required in BIDS.
+raw:
+  code:
+    name: code
+    level: optional
+    opaque: true
+  derivatives:
+    name: derivatives
+    level: optional
+    opaque: true
+  phenotype:
+    name: phenotype
+    level: optional
+    opaque: false
+  sourcedata:
+    name: sourcedata
+    level: optional
+    opaque: true
+  stimuli:
+    name: stimuli
+    level: optional
+    opaque: true
+  subjects:
+    entity: subject
+    level: required
+    opaque: false
+    subdirs:
+      sessions:
+        entity: subject
+        level: optional
+        opaque: false
+        subdirs:
+          datatype:
+            value: datatype
+            level: required
+            opaque: false
+      datatype:
+        value: datatype
+        level: optional
+        opaque: false
+
+alt-raw:
```

Comment:
`alt-raw` makes more sense to me, though maybe `validate` instead of `opaque`?

Reply:
I'm not committed to opaque, but validate seems to be a bit more of a directive than a description of the directory.